content/engine/swarm/secrets.md

```diff
@@ -748,11 +748,6 @@ line.
     wvnh0siktqr3  mysql  replicated  1/1       mysql:latest
     ```
 
-    At this point, you could actually revoke the `mysql` service's access to the
-    `mysql_password` and `mysql_root_password` secrets because the passwords
-    have been saved in the MySQL system database. Don't do that for now, because
-    we use them later to facilitate rotating the MySQL password.
-
```

Comment:
Do we know why the error occurred that was mentioned in the ticket? I _think_ that password may indeed not be needed after initialisation, or at least, missing that file should not cause the container to fail; also looking at the entry point-script at line 39; I _think_ it should handle missing files /cc @tianon

https://github.com/docker-library/mysql/blob/3e959c224b965b0dd92a59a1dedeb7c34a24f550/8.0/docker-entrypoint.sh#L38-L40

```bash
elif [ "${!fileVar:-}" ]; then
	val="$(< "${!fileVar}")"
fi
```

Does it actually _fail_ or is the "error" only due to the detection (so more an "informational message" - "we checked if the file exists, and it doesn't")?

```
/usr/local/bin/docker-entrypoint.sh: line 39: /run/secrets/mysql_password: No such file or directory
```

Reply:
Checked again, and for the mysql service, it errors out:

```console
$ docker service update --secret-rm mysql_password mysql
mysql
overall progress: 1 out of 1 tasks 
1/1: running   [==================================================>] 
verify: Waiting 5 seconds to verify that tasks are stable... 
service update paused: update paused due to failure or early termination of task rsmkmcu5hippb67p5upjd7
```

But it seems that the wordpress service doesn't, it continues running and with access to the database. I got an error when I tested it last time, but that was probably due to the permission mode 0400 error, where the secret mount wasn't readable by the wordpress user.

I can reinstate this statement for the wordpress service, but it does seem to apply for the mysql service.

Reply:
Thanks! Yeah, I'd have to re-read all things within context. The intent here may have been to pass a secret for _initializing_ (initial, temporary password), then update the service to remove the temporary password after the _real_ password has been configured.

But it's honestly quite some time since I looked at it (or used mysql containers) so would have to re-read 🙈